Source/Scene/Cesium3DTile.js

```diff
@@ -0,0 +1,1018 @@
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Number} The distance, in meters, or zero if the camera is inside the bounding volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.distanceToTileCenter = function(frameState) {
+        var boundingVolume = getBoundingVolume(this, frameState).boundingVolume;
+        var toCenter = Cartesian3.subtract(boundingVolume.center, frameState.camera.positionWC, scratchCartesian);
+        var distance = Cartesian3.magnitude(toCenter);
+        Cartesian3.divideByScalar(toCenter, distance, toCenter);
+        var dot = Cartesian3.dot(frameState.camera.directionWC, toCenter);
+        return distance * dot;
+    };
+
+    /**
+     * Checks if the camera is inside the viewer request volume.
+     *
+     * @param {FrameState} frameState The frame state.
+     * @returns {Boolean} Whether the camera is inside the volume.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.insideViewerRequestVolume = function(frameState) {
+        var viewerRequestVolume = this._viewerRequestVolume;
+        if (!defined(viewerRequestVolume)) {
+            return true;
+        }
+
+        return (viewerRequestVolume.distanceToCamera(frameState) === 0.0);
+    };
+
+    var scratchMatrix = new Matrix3();
+    var scratchScale = new Cartesian3();
+    var scratchHalfAxes = new Matrix3();
+    var scratchCenter = new Cartesian3();
+    var scratchRectangle = new Rectangle();
+
+    /**
+     * Create a bounding volume from the tile's bounding volume header.
+     *
+     * @param {Object} boundingVolumeHeader The tile's bounding volume header.
+     * @param {Matrix4} transform The transform to apply to the bounding volume.
+     * @param {TileBoundingVolume} [result] The object onto which to store the result.
+     *
+     * @returns {TileBoundingVolume} The modified result parameter or a new TileBoundingVolume instance if none was provided.
+     *
+     * @private
+     */
+    Cesium3DTile.prototype.createBoundingVolume = function(boundingVolumeHeader, transform, result) {
+        var center;
+        if (defined(boundingVolumeHeader.box)) {
+            var box = boundingVolumeHeader.box;
+            center = Cartesian3.fromElements(box[0], box[1], box[2], scratchCenter);
+            var halfAxes = Matrix3.fromArray(box, 3, scratchHalfAxes);
+
+            // Find the transformed center and halfAxes
+            center = Matrix4.multiplyByPoint(transform, center, center);
+            var rotationScale = Matrix4.getRotation(transform, scratchMatrix);
+            halfAxes = Matrix3.multiply(rotationScale, halfAxes, halfAxes);
+
+            if (defined(result)) {
+                result.update(center, halfAxes);
+                return result;
+            }
+            return new TileOrientedBoundingBox(center, halfAxes);
+        } else if (defined(boundingVolumeHeader.region)) {
```

Comment:
No need for these to be `else if` since they all return at the end.  Just use separate `if` calls.